src/transformers/modeling_tf_utils.py

```diff
@@ -952,6 +979,47 @@ def load_tf_weights(model, resolved_archive_file, ignore_mismatched_sizes=False,
     return missing_layers, unexpected_layers, mismatched_layers
 
 
+def load_tf_weights_from_safetensors(model, resolved_archive_file, ignore_mismatched_sizes=False, _prefix=None):
+    # Read the safetensors file
+    state_dict = safe_load_file(resolved_archive_file)
+
+    weight_value_tuples = []
+    mismatched_layers = []
+
+    weight_names = [format_weight_name(w.name, _prefix=_prefix) for w in model.weights]
+    loaded_weight_names = list(state_dict.keys())
+
+    # Find the missing layers from the high level list of layers
+    missing_layers = list(set(weight_names) - set(loaded_weight_names))
+    # Find the unexpected layers from the high level list of layers
+    unexpected_layers = list(set(loaded_weight_names) - set(weight_names))
+
+    weight_value_tuples = []
+    for weight in model.weights:
+        weight_name = format_weight_name(weight.name, _prefix=_prefix)
+        if weight_name in state_dict:
+            weight_value = state_dict[weight_name]
+            # Check if the shape of the current weight and the one from the H5 file are different
+            if K.int_shape(weight) != weight_value.shape:
```

Comment:
```suggestion
            if tuple(weight.shape.as_list()) != weight_value.shape:
```
I realize `K.int_shape` gets you a shape tuple quickly but it feels a bit like backend black magic, that method isn't well documented anywhere. I prefer the standard `weight.shape.as_list()` to get a shape list and then just converting that to `tuple`.

@gante do you have any preference here?

Reply:
(This also applies to the other uses of `int_shape` in this method)

Reply:
This is copied over from other code mind you, I did not invent this.
To minimize disruption maybe replace all instances of `K.int_shape` after this is merged in a followup PR?

Reply:
Works for me!